### PR TITLE
chore(ItemLayout): Convert to RFC

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

# BREAKING CHANGES

Converting `ItemLayout` from class component to functional. Any props are not longer passed to styles function. Removed `state` parameter from `renderContentArea`, `renderHeaderArea` and `renderMainArea`

Related to #12237

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/ItemLayout/ItemLayout.tsx [hunk 1]
```diff
@@ -34,9 +34,9 @@ export interface ItemLayoutProps extends UIComponentProps, ContentComponentProps
   endMedia?: any;
   headerMedia?: any;
   media?: any;
-  renderContentArea?: (props: ItemLayoutProps, state: any, classes: ComponentSlotClasses) => React.ReactNode;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/ItemLayout/ItemLayout.tsx:37]: This looks like a breaking change, please mention this.

**Comment by layershifter** [comment]
> Any props are not longer passed to styles function.

It's not true, at least `debugLayout`.

**Comment by assuncaocharles** [comment]
> > Any props are not longer passed to styles function.
> 
> It's not true, at least `debugLayout`.

But it doesn't come from props. For what I understand isn't for development purposes?

**Comment by layershifter** [comment]
> > > Any props are not longer passed to styles function.
> > 
> > 
> > It's not true, at least `debugLayout`.
> 
> But it doesn't come from props. For what I understand isn't for development purposes?

OMG, I see 🦀

**Comment by assuncaocharles** [comment]
> > > > Any props are not longer passed to styles function.
> > > 
> > > 
> > > It's not true, at least `debugLayout`.
> > 
> > 
> > But it doesn't come from props. For what I understand isn't for development purposes?
> 
> OMG, I see 🦀

I can even remove it at all if you think it would be better.

File: packages/fluentui/react-northstar/src/components/ItemLayout/ItemLayout.tsx (hunk 1 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/ItemLayout/ItemLayout.tsx:37]: It wasn't used at all anyway before, should I still mention?

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/ItemLayout/ItemLayout.tsx:37]: Mentioned

File: packages/fluentui/react-northstar/src/components/ItemLayout/ItemLayout.tsx [hunk 2]
```diff
@@ -64,151 +64,183 @@ export const itemLayoutSlotClassNames: ItemLayoutSlotClassNames = {
...
+  headerCSS: PropTypes.object,
+  headerMediaCSS: PropTypes.object,
+  contentCSS: PropTypes.object,
+  contentMediaCSS: PropTypes.object,
+  endMediaCSS: PropTypes.object,
+};
 
-  static deprecated_className = itemLayoutClassName;
+ItemLayout.defaultProps = {
+  as: 'div',
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/ItemLayout/ItemLayout.tsx:171]:
```suggestion
```

**Comment by layershifter** [comment]
> I can even remove it at all if you think it would be better.

If it's impossible to use, let's just remove these styles and props 👍

**Pull request merged**